### DISCUSSION TITLE
JSON configuration of Sonoff 4CH Pro with inverted switch state

### DISCUSSION THREAD

**Discussion by Roxirious** [RavenSystem/esp-homekit-devices#1723] [MEPLHAA Scripts]
Hi, can someone help with the json configuration of Sonoff 4CH Pro?
I need to invert the state of one of the relays in the Home app. When the relay is OFF, the state to be ON and vice versa. The idea is that this device is connected to the “normally closed” terminal, so even if there is no power supplied to the Sonoff, the device will still be powered ON. Basically, if the relay is OFF, state in Home app will be ON and if it’s ON, state will be OFF. Is this possible and if yes, how?
I started with the default json for this Sonoff in the “Device database”, but got lost. Any help will be appreciated!

**Comment by mariusvasss** [comment]
Invert the state of gpio needed.

"0":{"r":[{"g":12,"v":1}]}, "1":{"r":[{"g":12,"v":0}]}

Do this on your line, it's just an example for you to notice what you have to change. 

You don't need to remove the device, and no needs to reset Homekit ID. 
Just edit your line and press save.

**Reply by Roxirious** [reply]
Hey, thanks for the help, but this is not what I was searching for. This approach switches the specified relay ON, on boot. I want it to be OFF, only to appear ON in the Home app.

**Reply by mariusvasss** [reply]
Then do this

"1":{"r":[{"g":12}]}, "0":{"r":[{"g":12,"v":1}]}

**Reply by Roxirious** [reply]
Same result.. Already tried all combinations on this line. This is my conf:

{"c":
    {"l":13, "b":[{"g":0,"t":5}]},
        "a":[    
        {"1":{"r":[{"g":15}]},"0":{"r":[{"g":15,"v":1}]},"b":[{"g":14}]},
        {"1":{"r":[{"g":12}]},"0":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]}, 
        {"1":{"r":[{"g":5}]},"0":{"r":[{"g":5,"v":1}]},"b":[{"g":9}]},
        {"0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1}]},"b":[{"g":10}]}
        ]
}

Anyway, thanks!

**Reply by nusf** [reply]
Try to use “f”:0 and “f”:1
